src/main/java/bio/terra/cli/command/config/set/Resources.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.cli.command.config.set;
+
+import bio.terra.cli.command.helperclasses.BaseCommand;
+import bio.terra.cli.context.GlobalContext;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+/** This class corresponds to the fourth-level "terra config set resources" command. */
+@Command(
+    name = "resources",
+    description = "Set the maximum number of resources allowed per workspace.")
+public class Resources extends BaseCommand {
+
+  @CommandLine.ArgGroup(exclusive = true, multiplicity = "1")
+  ImageIdArgGroup argGroup;
+
+  static class ImageIdArgGroup {
+    @CommandLine.Option(
+        names = "--max",
+        description = "maximum number to allow before throwing an error")
```

Comment:
"throwing an error" sounds more programmer-y than user-y. Can a user still work with the first N resources happily, or is the order indeterminate?

Reply:
No, the user can't work with a subset of the resources, at least not with the code currently. The reason is that we pass a map of resource -> resource cloud identifier (e.g. `mybucket` -> `gs://my-bucket`) to all tools via environment variables. If we have so many resources (i.e. > max cache size) that they no longer fit comfortably on disk, then that approach seems less viable and we need to consider an overhaul in that area.